src/services/MetricsService/FilteredMetricsService.ts

```diff
@@ -0,0 +1,193 @@
+          }
+          break;
+        case EventTypeMetric.OtkWithDate:
+          return this.publishEvent(EventTypeMetric.OtkWithDate, []);
+        case EventTypeMetric.EnToggle:
+          return this.publishEnToggleEventIfNecessary(eventWithContext.state, eventWithContext.onboardedDate);
+        case EventTypeMetric.ExposedClear:
+          if (eventWithContext.exposureStatus.type === ExposureStatusType.Exposed) {
+            return this.publishEvent(EventTypeMetric.ExposedClear, [
+              [
+                'hoursSinceExposureDetectedAt',
+                String(getHoursBetween(getCurrentDate(), new Date(eventWithContext.exposureStatus.exposureDetectedAt))),
+              ],
+            ]);
+          }
+          break;
+      }
+
+      return Promise.resolve();
+    });
+  }
+
+  sendDailyMetrics(systemStatus: SystemStatus, notificationStatus: Status): Promise<void> {
+    return this.serialPromiseQueue.add(() => {
+      return this.publishOnboardedEventIfNecessary(
+        String(notificationStatus),
+        String(systemStatus === SystemStatus.Active),
+      ).then(() => this.metricsService.sendDailyMetrics());
+    });
+  }
+
+  private publishInstalledEventIfNecessary(): Promise<void> {
+    const publishAndMark = (): Promise<void> => {
+      return this.publishEvent(EventTypeMetric.Installed, [], true).then(() =>
+        this.stateStorage.markInstalledEventAsPublished(),
+      );
+    };
+    return this.stateStorage.isInstalledEventPublished().then(isPublished => {
+      if (isPublished === false) {
+        return publishAndMark();
+      }
+    });
+  }
+
+  private publishOnboardedEventIfNecessary(notificationStatus: string, frameworkenabled: string): Promise<void> {
+    const publishAndMark = (): Promise<void> => {
+      return this.publishEvent(EventTypeMetric.Onboarded, [
+        ['pushnotification', notificationStatus],
+        ['frameworkenabled', frameworkenabled],
+      ]).then(() => this.stateStorage.markOnboardedEventShouldNotBePublished());
+    };
+    return this.stateStorage.shouldOnboardedEventBePublished().then(shouldPublish => {
+      if (shouldPublish) {
+        return publishAndMark();
+      }
+    });
+  }
+
+  private publishEnToggleEventIfNecessary(state: boolean, onboardedDate: Date | undefined): Promise<void> {
+    function shouldPublishEnToggle(): boolean {
+      if (state) {
+        if (onboardedDate && getHoursBetween(onboardedDate, new Date()) > 24) {
+          return true;
+        } else {
+          return false;
+        }
+      } else {
+        return true;
+      }
+    }
+
+    if (shouldPublishEnToggle()) {
+      return this.publishEvent(EventTypeMetric.EnToggle, [['state', String(state)]]);
+    } else {
+      return Promise.resolve();
+    }
+  }
+
+  private publishEvent(
+    eventType: EventTypeMetric,
+    metricPayload: [string, string][],
+    forcePushToServer = false,
+  ): Promise<void> {
+    return this.getRegion().then(region => {
```

Comment:
Would prefer await syntax here vs the then

Reply:
This is a discussion we had with @jeberhardt yesterday. I strongly like the functional approach so I guess it is something we can discuss during the review. Why is there only two places where you would like to use await instead of then in this pull request?

Reply:
We can discuss in the review 

I would prefer to use `await` in all cases but lets discuss